exp/api/v1alpha1/azureasomanagedcluster_types.go

```diff
@@ -35,6 +35,22 @@ type AzureASOManagedClusterSpec struct {
 
 // AzureASOManagedClusterStatus defines the observed state of AzureASOManagedCluster.
 type AzureASOManagedClusterStatus struct {
+	//+optional
+	Resources []ResourceStatus `json:"resources,omitempty"`
```

Comment:
The 'embedded in a CAPZ resource" is specifically to support users configuring features of their ManagedCluster that CAPZ itself doesn't yet support by going to "the source", is that right?

Reply:
In fact this is an entirely new "ASO-native" CAPZ API that will delegate the entire AKS API to ASO. So "CAPZ support == ASO support" in terms of AKS features.

Due to the (correct) strongly versioned and fast moving ASO API type system, embedding a literal strongly typed ASO data structure is not going to be an ideal option, so this more loosely typed implementation (essentially relies upon RAW JSON equivalency between these new CAPZ resource types and the underlying ASO types is a path where CAPZ and ASO can move forward independently with optimal freedom (CAPZ's release velocity isn't strictly tied to ASO's release velocity.

@nojnhuh can add more color here as well

Reply:
Is your specific curiosity the status reporting here? I don't fully grok the expected user scenarios here, but happy to chat either here, in Teams or in a quick call. If the user "knows" this is ASO and is expected to mostly look at the underlying ASO resources for detailed status then what you have here seems reasonable from a status reporting perspective. OTOH if you're trying to convey more detailed information from the underlying ASO resources here you're missing a lot because there's a `Reason` attached to each Condition in ASO which would indicate the cause of the "problem" (if there is one), which users won't be able to see with a simple boolean representation like you've got here.

Reply:
> If the user "knows" this is ASO and is expected to mostly look at the underlying ASO resources for detailed status then what you have here seems reasonable from a status reporting perspective.

This is what I've been thinking so far. While this is still very alpha, I was thinking of keeping the status only to what CAPZ needs to operate but being open to expanding it later. I can see the RBAC reasons for why we wouldn't want to mirror the entire status of each ASO resource in its CAPZ resource.

Reply:
"What CAPZ needs to operate" is a pretty reasonable line to draw, especially to start. Starting with less and then adding more if there's some really compelling reason to do so makes a lot of sense.